determined-ai/determined
# feat: Launch JupyterLab" button should appear on tasks page [WEB-811]

## Description

feat: Launch JupyterLab" button should appear on tasks page [WEB-811]

Updates:

1. Added `canCreateNSC` and `canCreateWorkspaceNSC` methods.
2.  Added required `Workspace` dropdown to the JupyterLab "Full Configuration" form.
3.  Added required `Workspace` dropdown to the JupyterLab "Simple Configuration" form.
4. Added ability to specify a workspace in `useModalJupyterLab` that will disable and preselect a value for  the dropdown in (2) and (3).
5. Removed the feature flag check that was previously hiding the `Launch JupyterLab` button on the `Tasks` page.

Note: The regular `Launch JupyterLab button` on the navigation sidebar has remained unchanged, in terms of being disabled/enabled. If we also need to change that buttons behavior then that can be done.





## Test Plan

1. Navigate to the `Tasks` page. 
2. Click the "Launch JupyterLab" button. 
3. Click `Launch` without specifying a workspace. Ensure you see the error message `Workspace is required`.
4. Click the Workspace dropdown and choose a workspace. 
5. Click "Launch" and ensure you are taken to the interactive tasks page. 
6. Navigate to the `Tasks` page. 
7.  Click the "Launch JupyterLab" button. 
8. Click the "Show Full Config" button. 
9. Click `Launch` without specifying a workspace. Ensure you see the error message `Workspace is required`.
10. Click the Workspace dropdown and choose a workspace. 
11. Click "Launch" and ensure you are taken to the interactive tasks page. 
12. In both the "Simple" and "Full" configuration forms ensure that changing values and submitting the form works as expected.
13. Navigate to a workspace page then the "Tasks" tab for the workspace. Click the "Launch JupyterLab" button. Ensure that both in the "Full" and "Simple" configuration the Workspace drowpdown is disabled and preselected to the current workspace. Click "Launch" and ensure the task shows up in the workspaces' "Tasks" list.
15. (Easiest tested with RBAC enabled) Navigate to the Tasks page or to a workspace where you do not have NSC permissions. Ensure the `Launch JupyterLab` button is disabled and a tooltip is shown.

![4CC3AA16-3CE3-4A2F-B811-733EAB10F038](https://user-images.githubusercontent.com/103522725/214357254-57027b53-6650-4673-8727-6f6838e4be2a.jpeg)





## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket

WEB-811 







[WEB-811]: https://determinedai.atlassian.net/browse/WEB-811?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
